core/src/main/java/org/apache/iceberg/io/FileWriter.java

```diff
@@ -54,6 +54,20 @@ default void write(Iterable<T> rows) {
    */
   void write(T row);
 
+  /**
+   * Returns the file path that are currently opened.
+   *
+   * @return the current file path.
+   */
+  CharSequence location();
+
+  /**
+   * Returns the row offset that are currently writing, starting from 0.
```

Comment:
```suggestion
   * Returns the row offset that will be written to next, starting from 0.
```
Is this interface intended to be seekable at any point?  If not maybe call this rowsWritten which might be clearer?

Reply:
As we've discussed in https://github.com/apache/iceberg/pull/4264/files#r819238913, I agree `recordCount` is a good & consistent name.

Reply:
As we've discussed in https://github.com/apache/iceberg/pull/4264/files#r819238913, I agree `recordCount` is a good & consistent name.